src/dmd/link.d

```diff
@@ -569,36 +569,23 @@ public int runLINK()
             argv.push("-Xlinker");
             argv.push("--gc-sections");
         }
-        /* Add each library, prefixing it with "-l".
+        /* Add each library, prefixing it with "-l" if necessary.
          * The order of libraries passed is:
-         *  1. any libraries passed with -L command line switch
-         *  2. libraries specified on the command line
+         *  1. static libraries ending with *.a
+         *  2. link switches passed with -L command line switch
          *  3. libraries specified by pragma(lib), which were appended
          *     to global.params.libfiles.
-         *  4. link switches, that may also contain -l libraries
-         *  5. standard libraries.
+         *  4. standard libraries.
          */
-        for (size_t i = 0; i < global.params.libfiles.dim; i++)
+        foreach (i; 0 .. global.params.libfiles.dim)
```

Comment:
If you are going the foreach way;
```D
foreach (p; global.params.libfiles)
```

Reply:
Didn't know it was possible with Array.
Still don't see what makes it legal. Is it opSlice? This isn't documented.